api-registration/server/server.ts

```diff
@@ -0,0 +1,151 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import 'dotenv/config';
+import pg from 'pg';
+import argon2 from 'argon2';
+import express from 'express';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+type User = {
+  username: string;
+  password: string;
+};
+type Entry = {
+  entryId: number;
+  title: string;
+  notes: string;
+  photoUrl: string;
+};
+
+const db = new pg.Pool({
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+app.use(express.json());
+
+app.post('/api/auth/sign-up', async (req, res, next) => {
+  try {
+    const { username, password } = req.body as Partial<User>;
+    if (!username || !password) {
+      throw new ClientError(400, 'username and password are required fields');
+    }
+    /* TODO:
+     * Hash the user's password with `argon2.hash()`
+     * Insert the user's "username" and "hashedPassword" into the "users" table.
+     * Respond to the client with a 201 status code and the new user's "userId", "username", and "createdAt" timestamp.
+     * Catch any errors.
+     *
+     * Hint: Insert statements can include a `returning` clause to retrieve the insterted row(s).
+     */
+    const hashedPassword = await argon2.hash(password);
+    const sql = `
+          insert into "users" ("username", "hashedPassword")
+            values($1, $2)
+          returning *;
+          `;
+    const params = [username, hashedPassword];
+    const result = await db.query(sql, params);
+    const newUser = result.rows[0];
+    delete newUser.hashedPassword;
```

Comment:
You can do it this way, or you can return just the columns you need in the `returning` clause. Or you can create just the object you want to return.

I find the `delete` approach the least desirable since you can still send back things you weren't intending to, especially if you later add columns to the table.